bundles/model/org.eclipse.smarthome.model.script/src/org/eclipse/smarthome/model/script/actions/BusEvent.java

```diff
@@ -140,14 +153,24 @@ static public Object postUpdate(String itemName, String stateString) {
             try {
                 Item item = registry.getItem(itemName);
                 State state = TypeParser.parseState(item.getAcceptedDataTypes(), stateString);
-                publisher.post(ItemEventFactory.createStateEvent(itemName, state));
+                if (state != null) {
+                    publisher.post(ItemEventFactory.createStateEvent(itemName, state));
+                } else {
+                    LoggerFactory.getLogger(BusEvent.class).warn(
+                            "Cannot convert '{}' to a state type which item '{}' accepts: {}.", stateString, itemName,
+                            getAcceptedDataTypeNames(item));
+                }
             } catch (ItemNotFoundException e) {
                 LoggerFactory.getLogger(BusEvent.class).warn("Item '{}' does not exist.", itemName);
             }
         }
         return null;
     }
 
+    private static <T extends State> List<String> getAcceptedDataTypeNames(Item item) {
```

Comment:
It is a pity that you have to double this method via copy&paste. Isn't there a common place for it that can be accessed from both classes?

Reply:
place it not the problem. Stupid java is:


"The method theCommondMethodThatWeAllWantedToWrite(List<Class<? extends Type>>) in the type BusEvent is not applicable for the arguments (List<Class<? extends Command>>)"